src/EventStore.Core/Services/Transport/Grpc/Enumerators.ReadAllBackwards.cs

```diff
@@ -43,96 +42,84 @@ public class ReadAllBackwards : IAsyncEnumerator<ResolvedEvent> {
 				}
 
 				_bus = bus;
-				_nextPosition = position;
 				_maxCount = maxCount;
 				_resolveLinks = resolveLinks;
 				_user = user;
 				_requiresLeader = requiresLeader;
 				_deadline = deadline;
-				_disposedTokenSource = new CancellationTokenSource();
-				_buffer = new ConcurrentQueue<ResolvedEvent>();
-				_tokenRegistration = cancellationToken.Register(_disposedTokenSource.Dispose);
-			}
+				_cancellationToken = cancellationToken;
+				_semaphore = new SemaphoreSlim(1, 1);
+				_channel = Channel.CreateBounded<ResolvedEvent>(BoundedChannelOptions);
 
+				ReadPage(position);
+			}
 
 			public ValueTask DisposeAsync() {
-				_disposedTokenSource.Dispose();
-				_tokenRegistration.Dispose();
-				return default;
+				_channel.Writer.TryComplete();
```

Comment:
@thefringeninja should we TryComplete() with an ObjectDisposedException here? e.g if someone calls MoveNextAsync() after DisposeAsync() (please also see: https://github.com/EventStore/EventStore/pull/2712/files?diff=split&w=1#r504469892)